src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -312,6 +312,19 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 			}
 			$theme_support_data['settings']['color']['defaultGradients'] = $default_gradients;
 
+			if ( ! isset( $theme_support_data['settings']['shadow'] ) ) {
+				$theme_support_data['settings']['shadow'] = array();
+			}
+			$default_shadows = false;
+			if ( current_theme_supports( 'default-shadow-presets' ) ) {
+				$default_shadows = true;
+			}
+			if ( ! isset( $theme_support_data['settings']['shadow']['presets'] ) ) {
+				// If the theme does not have any shadows, we still want to show the core ones.
+				$default_shadows = true;
+			}
```

Comment:
Just a quick comment, as I haven't looked at this change in detail, but this line stood out to me while passing by. Does this check mean that if the theme doesn't add `current_theme_supports( 'default-shadow-presets' )` it'll still wind up getting the default shadow presets?

Reply:
Yes, that's correct.

I think from all of the discussions that I read, the desire was to go for consistency with other settings, so I copied the behavior from colors and gradients. However, I know that behavior is different from what you had talked about in https://github.com/WordPress/gutenberg/issues/58908.

If we want to match the `appearance-tools` behavior that you were going for there, this can be moved just a few lines lower into the `appearance-tools` section.

Reply:
I don't feel too strongly about it, but mostly trying to follow the general discussion from #58908. My impression was that shadows / effects are a bit of a special case, so guarding behind `appearance-tools` is likely a better fit so that we're not outputting anything for themes that don't explicitly opt-in either to the presets or to `appearance-tools`, but @carolinan might have more insight for the Classic themes use case than me here 🙂

Reply:
And fair warning, I haven't really tested any of this code yet. I wanted to get something pushed up as a starting point that folks could look at. But it's all copied from colors/gradients configuration, so I'm relatively confident that it should be working.

Reply:
> so that we're not outputting anything for themes that don't explicitly opt-in either to the presets

This is often a point of confusion for global styles, but setting `default*` options to `false` does not prevent the styles from being output. What I've been told is that the default styles always need to exist in order to support patterns between themes. The only preset that handles this differently is duotone.

Reply:
Ah, gotcha.

> The only preset that handles this differently is duotone.

I suppose that's the question then: whether shadows are like colors or font sizes, or whether they're more like duotone in that respect. It might be worth getting more opinions on this? I don't feel strongly either way, my main hope is that whichever way it works feels natural and not unexpected to folks. Maybe @madhusudhand might have some thoughts, too?

Reply:
> I suppose that's the question then: whether shadows are like colors or font sizes, or whether they're more like duotone in that respect.

The only reason duotone is different is because of the large SVGs that were also being generated https://github.com/WordPress/gutenberg/issues/38299. Changing it was not an easy feat, and there are still unresolved issues from making that change https://github.com/WordPress/gutenberg/issues/53662.

For something small like a few lines of shadow presets, I promise it isn't worth it.

Reply:
Good to know, thanks!